cursor_pr_review.py

```
@@ -233,7 +273,233 @@ def _validate_anthropic(self) -> Dict[str, Any]:
+                }
+            )
+            response.raise_for_status()
+
+            result = response.json()
+            analysis = result['choices'][0]['message']['content']
+
+            # Parse analysis into review comments
+            return self._parse_ai_analysis(analysis)
+
+        except requests.exceptions.RequestException as e:
+            raise APIError(f"OpenAI analysis failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)